src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -455,6 +430,44 @@ private Collection<FunctionDescriptor> ReadFunctions(ScriptHostConfiguration con
             return ReadFunctions(metadatas, descriptorProviders);
         }
 
+        /// <summary>
+        /// Determines which script should be considered the "primary" entry point script.
+        /// </summary>
+        internal static string DeterminePrimaryScriptFile(JObject functionConfig, string[] functionFiles)
+        {
+            if (functionFiles.Length == 1)
+            {
+                // if there is only a single file, that file is primary
+                return functionFiles[0];
```

Comment:
It's debatable. My take is that we cause more harm with the 'second source file cliff' than if you force people to follow a (simple) pattern from the get go.